workflow/rules/common.smk

```diff
@@ -1,29 +1,73 @@
 import re
 import logging
 import sys
+import pysam
 
 FIRST_REPORT = True
 
 
-def get_ref():
+def get_ref_orig():
     if "ref" not in config:
         raise ValueError("FIRE: ref parameter is missing in config.yaml")
     ref = config["ref"]
     if not os.path.exists(ref):
         raise ValueError(f"FIRE: reference file {ref} does not exist")
     return os.path.abspath(ref)
 
+def get_ref_old(wc):
+    if "ref" in config:
+        ref = config["ref"]
+    if "ref" not in config:
+        if "ref" in MANIFEST.columns:
+            ref = get_input_ref()
+        else: 
+            raise ValueError("FIRE: ref parameter is missing in config.yaml and no ref column in manifest")
+    #ref = config["ref"]
+    if not os.path.exists(ref):
+        raise ValueError(f"FIRE: reference file {ref} does not exist")
+    return os.path.abspath(ref)
 
-def get_fai():
-    fai = f"{get_ref()}.fai"
+def get_ref(wc):
+    ref = MANIFEST.loc[wc.sm, "ref"]
+    if not os.path.exists(ref):
+        raise ValueError(f"FIRE: reference file {ref} does not exist")
+    return os.path.abspath(ref)
+
+def get_ref_name_old(wc):
+    if "ref_name" in config:
+        ref_name=config["ref_name"]
+    else:
+        if "ref_name" in MANIFEST.columns:
+            ref_name= get_input_ref_name()
+        else:
+            raise ValueError("FIRE: ref_name parameter is missing in config.yaml and no ref_name column in manifest")
+    #ref=get_ref()
+    #temp_split=ref.strip().split('/')[-1]
+    #fa_split=temp_split.split('.fa')
+    #ref_name = '.fa'.join(fa_split[:-1])
+    return(ref_name)
+
+def get_ref_name(wc):
+    return MANIFEST.loc[wc.sm, "ref_name"]
+
+def get_fai_orig(wc):
+    fai = f"{get_ref(wc)}.fai"
     if not os.path.exists(fai):
         raise ValueError(f"FIRE: reference index file {fai} does not exist")
     return fai
 
+def get_fai(wc):
+    ref = MANIFEST.loc[wc.sm, "ref"]
```

Comment:
to get `ref` you should call `get_ref`. And then generally it is good to use f strings in situations like this:
```
ref = get_ref(wc)
fai = f"{ref}.fai"
...
```